rbac/default-role-manager/role_manager.go

```diff
@@ -529,10 +535,18 @@ func (dm *DomainManager) Match(str string, pattern string) bool {
 }
 
 func (dm *DomainManager) rangeAffectedRoleManagers(domain string, fn func(rm *RoleManagerImpl)) {
-	if dm.domainMatchingFunc != nil {
+	if dm.domainMatchingFunc != nil || dm.domainHierarchyManager != nil {
 		dm.rmMap.Range(func(key, value interface{}) bool {
 			domain2 := key.(string)
-			if domain != domain2 && dm.Match(domain2, domain) {
+			matched := false
```

Comment:
This can probably be packed similarly to [this](https://github.com/casbin/node-casbin/pull/380/files#diff-d6c8111bf43f43510834bf57fa2402634aeec52b3988dd17230dfe3cdb7baba5R202).

Otherwise lgtm.